packages/interfaces/src/method-invocation-message.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2011-2018 Genestack Limited
+ * All Rights Reserved
+ * This source code is distributed under the MIT license:
+ * https://github.com/genestack/JS-SDK/tree/master/LICENSE
+ */
+
+import {JSONValue} from './json-value';
+import SystemMessage from './system-message';
+
+type MethodInvocationMessageType = 'G$:invoke-method';
+
+export const METHOD_INVOCATION_MESSAGE_TYPE: MethodInvocationMessageType = 'G$:invoke-method';
+
+type ApplicationId = string; // should be regex e.g. /w+\/w+/ probably,
+                             // if https://github.com/Microsoft/TypeScript/issues/6579 resolved
+
+/**
+ * @property applicationId      Id of application in format "vendor/application"
+ * @property method             Application method name
+ * @property parameters         A list of parameters that method accepts
+ * @property showBusyIndicator  Whether a global busy indicator should be shown
+ * @property extendSession      If true, this method call will extend current user session
```

Comment:
Here I just tried out how TypeDoc works in IDE. For now we're gonna use hand-crafted markdown readmes.